### PR TITLE
Fixes #348 Correct hyperlinks are displayed

### LINKED ISSUE #348: Problem in search results, no hyperlinks displayed.
For certain search queries, the results displayed contain no hyperlink to click on but read only links.
issue

### PR DESCRIPTION
**Pull request by harshit98**
Fixes issue #348 

Changes: 
- Added code to filter out the links with `jpg` format in standard results section.
- Added comments to code, to review PR faster and it would be easier for new contributors to contribute in Susper project and understand the code.

Demo Link: <a href="https://harshit98.github.io/susper.com/">here</a>.

Screenshots for the change: 

Before -
screenshot from 2017-06-02 02-28-35

After -
screenshot from 2017-06-02 02-30-14

Kindly review @nikhilrayaprolu @Marauderer97

### CODE REVIEW

**Comment by harshit98** [comment]
@sandeepm96 yes. Just like Google standard results work. There were some data with jpg formats, which I have filtered them out within standard results only. For png formats I didn't find any link regarding to it.

**Comment by sandeepm96** [comment]
@harshit98 But why show the name of the file. The result should be completely filtered out. I mean it should not be displayed at all.

**Comment by harshit98** [comment]
@sandeepm96 #348 in this issue, see the screenshot. :)

On YaCy server already work is going on. It always get updated. I have the done the work according to it as shown in screenshot. This error was reproduced by typing query `hello` . So, at present on typing that query, I was getting the results with `jpg` formats. Just to show, issue has been solved I have typed query `.jpg` . If you feel, anything more has to be done, I'll open up a separate issue for that. :)

**Comment by harshit98** [comment]
@isuruAb That won't be happening. See issue #348 , I have done work according to it. Server gets updated. The screenshots here are just to show reproduce that bug, and it's fix. :)

@nikhilrayaprolu Okay. I'll remove the unnecessary comments and will go through your suggestions. If there won't be any syntax for filtering it out, I'll let you know. :)

**Comment by mariobehling** [comment]
Please also resolve merge conflicts.

**Comment by harshit98** [comment]
@mariobehling conflicts has been resolved. :+1:

@nikhilrayaprolu @Marauderer97 @sandeepm96 @isuruAb Kindly review, I have done some more changes. 

Screenshots -
screenshot from 2017-06-04 08-54-58

screenshot from 2017-06-04 08-55-39

on searching of jpg file, all the results are being filtered out containing jpg format. (This does not affects 'images' section)
screenshot from 2017-06-04 08-56-21

**Comment by nikhilrayaprolu** [comment]
@Orbiter is there any specific syntax to query Yacy for results excluding images?
I have an idea of using regex queries for implementations of this kind.

File: src/app/results/results.component.ts [hunk 1]
```diff
@@ -170,22 +173,38 @@ export class ResultsComponent implements OnInit {
...
     });
-
-
   };
 
   ngOnInit() {
 
   }
+
+  filterLink(link) {
```
Comment (nikhilrayaprolu) [inline src/app/results/results.component.ts]: excluding results should not be done on front end as the count of results per page decreases. In whatever situation we have to display 10 results per page. 
Okay to be clear,
suppose we have 10 initial results and all are jpg, this gives us zero results on first page,on which user assumes that there are no results supporting his search query.

**Comment by harshit98** [comment]
@nikhilrayaprolu Hmm. You're right. Excluding results should be done on front end.

**Comment by nikhilrayaprolu** [comment]
@harshit98 so ping the backend team and ask them how they could help you on this issue in gitter.

**Comment by harshit98** [comment]
Closing this PR, since it is a server issue. (See discussion on gitter)

**Pull request closed without merging**